cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -423,6 +414,11 @@ public boolean apply(Id.Program programId) {
   }
 
   private AppFabricServiceStatus removeApplication(final Id.Application appId) throws Exception {
+    // Deletion of a particular application is not allowed if that application is used by an adapter
```

Comment:
This logic is moved from here so that it gets executed for both the handlers of /apps/app-id DELETE and /apps DELETE